test/sig0-test.js

```diff
@@ -41,4 +49,44 @@ describe('SIG(0)', function() {
 
     assert(sig0.verify(signedRaw, key));
   });
+
+  it('should sign and verify message w/ privatedns', () => {
+    const msg = new Message();
+
+    msg.qr = true;
+    msg.ad = true;
+
+    msg.question.push(Question.fromString('example.com. IN A'));
+    msg.answer.push(Record.fromString('example.com. 300 IN A 172.217.0.46'));
+
+    const priv = hsig.createPrivate();
+    const pub = hsig.createPublic(priv);
+    const key = hsig.createKey(pub);
+
+    const signer = (priv, data) => {
+      return secp256k1.sign(data, priv);
+    };
+
+    const msgRaw = msg.compress();
+    const signedRaw = sig0.sign(msgRaw, key, priv, null, signer);
```

Comment:
Could you test `hsig.sign` and `hsig.verify`? Those are really the two important methods in practice

Reply:
Added. I had to update `hsig.verify` to parse the name + pubkey scheme.